presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -237,7 +240,8 @@ public void initialize(URI uri, Configuration conf)
         String userAgentPrefix = conf.get(S3_USER_AGENT_PREFIX, defaults.getS3UserAgentPrefix());
         this.skipGlacierObjects = conf.getBoolean(S3_SKIP_GLACIER_OBJECTS, defaults.isSkipGlacierObjects());
         this.s3StorageClass = conf.getEnum(S3_STORAGE_CLASS, defaults.getS3StorageClass());
-
+        this.webIdentityEnabled = conf.getBoolean(S3_WEB_IDENTITY_ENABLED, false);
+        checkArgument(!(webIdentityEnabled && isNullOrEmpty(s3IamRole)), "Invalid configuration: s3IamRole must be provided when webIdentityEnabled is set to true");
```

Comment:
nit: since this error is shown to the user, should we mention the actual config name rather than the variable name?

```suggestion
        checkArgument(!(webIdentityEnabled && isNullOrEmpty(s3IamRole)), "Invalid configuration: hive.s3.iam-role must be provided when hive.s3.web.identity.auth.enabled is set to true");
```